common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -183,6 +204,19 @@ public static int findInSet(final UTF8String match, final UTF8String set, int co
     return 0;
   }
 
+  public static int lowercaseIndexOf(final UTF8String target, final UTF8String pattern,
+      final int start) {
+    if (pattern.numChars() == 0) return 0;
```

Comment:
I would expect `start` to be returned here, not `0`.

Reply:
that's not expected behaviour for `indexOf` in general, so I think it should stay 0 for `lowercaseIndexOf`

UTF8String `public int indexOf`:
```
    if (v.numBytes() == 0) {
      return 0;
    }
```

CollationAwareUTF8String `public static int indexOf`:
```
    if (pattern.numBytes() == 0) {
      return 0;
    }
```

Reply:
I have to admit this is a bit strange. Can we check with the OSS team that the UTF8String behavior is the expected one?

Reply:
why do you think it would not be expected? `indexOf` uses 0-based index

- -1 means "not found"
- 0 means "found at beginning"

an empty pattern string is (by definition) found at the beginning of any given target string
I don't think there are any issues there - seems pretty expected to me